preview-maker.js

```diff
@@ -79,33 +101,127 @@ const makePrev = async (themeBuffer, themeName, themeAuthor, template) => {
+                colorsHue.push(colornow);
+            }
+        }
+        if (colorsHue.length > 1) {
+            var minDifference = 360
+            for (const colorHue of colorsHue) {
+                for (const hue of colorsHue) {
+                    if (hue != colorHue) {
+                        if (Math.abs(hue - colorHue) < minDifference) {
+                            minDifference = Math.abs(hue - colorHue)
+                            accentHue = (hue + colorHue) / 2
+                        }
+                    }
+                }
+            }
+        }
+    } else {
+        accentHue = rgbToHsl(theme['chats_actionBackground'] || defaultVariablesValues['chats_actionBackground']).hue
+    }
+    for (const outBubbleGradientelement of getElementsByClassName(preview, "chat_outBubbleGradient")){
+        const chat_outBubble = theme[`chat_outBubble`] || defaultVariablesValues["chat_outBubble"];
+        fill(outBubbleGradientelement, theme[`chat_outBubbleGradient`] || chat_outBubble);
+    }
+    for (const PreviewBackLinear of getElementsByClassName(preview, "PreviewBackLinear")) {
+        fillHSL(PreviewBackLinear, `hsl(${rgbToHsl(theme[`chat_outBubbleGradient`]).hue}, 100%, 90%)`);
+    }
+    for (const PreviewBackLinearShadow of getElementsByClassName(preview, "PreviewBackLinearShadow")) {
+        fillHSL(PreviewBackLinearShadow, `hsl(${rgbToHsl(theme[`chat_outBubble`] || defaultVariablesValues["chat_outBubble"]).hue}, 100%, 90%)`);
+    }
+    for (const PreviewBack of getElementsByClassName(preview, "PreviewBack")) {
+        fillHSL(PreviewBack, `hsl(${accentHue}, 100%, 90%)`);
+    }
+    for (const ChatShadow of getElementsByClassName(preview, "ChatShadow")) {
+        fillHSL(ChatShadow, `hsl(${accentHue}, 100%, 2%)`);
+    }
+    if (theme[`chat_outBubbleGradient`] != undefined) {
+        for (const PreviewBack of getElementsByClassName(preview, "PreviewBack")){
+            fill(PreviewBack, {red:0,green:0,blue:0,alpha:0});
+        }
+    }
+    for (const avatar of avatars){
+        const windowBackgroundWhite = theme['windowBackgroundWhite'] || defaultVariablesValues['windowBackgroundWhite']
+        if (Math.abs(theme[avatar].red - windowBackgroundWhite.red) + Math.abs(theme[avatar].green - windowBackgroundWhite.green) + Math.abs(theme[avatar].blue - windowBackgroundWhite.blue) < 15){
+            for (const ava of getElementsByClassName(preview, avatar)){
+                fill(ava, theme['avatar_text'] || defaultVariablesValues['avatar_text'])};
+                for (const avashadow of getElementsByClassName(preview, `${avatar}Shadow`)) {
+                    const choose = theme['avatar_text'] || defaultVariablesValues['avatar_text'];
+                    const hslChoose = rgbToHsl(choose)
+                    fillHSL(avashadow,`hsl(${hslChoose.hue}, ${hslChoose.saturation}%, ${hslChoose.lightness - hslChoose.lightness > 25 && rgbToHsl(windowBackgroundWhite).lightness < 10 ? -20 : 40 }%)`)
```

Comment:
Avoid long lines, they're hard to read. You can, for example, put temporary but meaningful expressions in their own variables.